lib/scumblr_tasks/security/curl_analyzer.rb

```diff
@@ -339,11 +339,16 @@ def request_metadata_parser(r, response, request_url, request_metadata, metadata
           else
             matched_expression = searched_code[1].to_s
           end
-          if r.metadata[:curl_metadata].key?(request_metadata.keys[line_no])
-            curl_metadata[request_metadata.keys[line_no] + "-" + duplicate_keys.to_s] = matched_expression.strip.truncate(300)
-            duplicate_keys += 1
+          if r.metadata[:curl_metadata].key?(request_metadata.keys[key_no])
+            curl_metadata[request_metadata.keys[key_no]] ||= []
+            curl_metadata[request_metadata.keys[key_no]] << matched_expression.strip.truncate(300)
+            curl_metadata[request_metadata.keys[key_no]].uniq!
+            curl_metadata[request_metadata.keys[key_no]]
           else
-            curl_metadata[request_metadata.keys[line_no]] = matched_expression.strip.truncate(300)
+            curl_metadata[request_metadata.keys[key_no]] ||= []
```

Comment:
The code in the else looks identical to that in the first part of the if... is this if/else necessary?